### PR TITLE
EVG-7627 trigger version from git tag (using yaml)

### PR DESCRIPTION
**Pull request by ablack12**
Users can either define project aliases the normal way, or using a new yaml file. If multiple git tag aliases match, then it has to be the normal way, since we shouldn't have to do any processing with the new yaml. Defining using aliases will be tested in another ticket. 

From staging test (pre author changes): https://evergreen-staging.corp.mongodb.com/version/yb_mci_testing1_5ed6deb69dbe320829565557

### CODE REVIEW

File: model/repository.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ var (
 
 type Revision struct {
 	Author          string
+	AuthorID        string
```
Comment (john-m-liu) [inline model/repository.go:45]: what's the difference between this and Author?

File: model/project_parser.go [hunk 2]
```diff
@@ -517,6 +520,24 @@ func LoadProjectInto(data []byte, identifier string, project *Project) (*ParserP
 	return intermediateProject, errors.Wrap(err, LoadProjectError)
 }
 
+func GetProjectFromFile(ctx context.Context, ref ProjectRef, file, token string) (*Project, *ParserProject, error) {
```
Comment (john-m-liu) [inline model/project_parser.go:523]: Is this mostly the same as the GetRemoteConfig function?

Comment (ablack12) [reply model/project_parser.go:523]: Yes 😮  I'll refactor

File: model/repository.go (hunk 1 above)
Comment (ablack12) [reply model/repository.go:45]: Author is the name "John Liu" while AuthorID is the Evergreen ID "john.liu"

**Pull request merged**